src/main/java/org/candlepin/model/DeletedConsumerCurator.java

```diff
@@ -117,6 +121,98 @@ public int countByConsumerUuid(String uuid) {
+        String principalName = (principal == null ? null : principal.getName());
+        OffsetDateTime now = OffsetDateTime.now();
+
+        // Get information about existing consumers to populate DeletedConsumer
+        String consumerJpql =
+            "SELECT c.id, c.uuid, c.ownerId, o.displayName, o.key, c.name " +
+            "FROM Consumer c " +
+            "JOIN c.owner o " +
+            "WHERE c.id IN (:consumerIds)";
+        List<Object[]> rows = this.getEntityManager()
+            .createQuery(consumerJpql, Object[].class)
+            .setParameter("consumerIds", consumerIds)
+            .getResultList();
+
+        if (rows.isEmpty()) {
+            return 0;
+        }
+
+        // All uuids that needs to be processed
+        List<String> uuids = new java.util.ArrayList<>(rows.stream().map(r -> (String) r[1]).toList());
+
+        // Get existing uuids from the DeletedConsumer table
+        String deletedConsumerUuidsJpql =
+            "SELECT dc.consumerUuid " +
+            "FROM DeletedConsumer dc " +
+            "WHERE dc.consumerUuid IN (:uuids)";
+        Set<String> existingUuids = new HashSet<>(
+            this.getEntityManager().createQuery(deletedConsumerUuidsJpql, String.class)
+                .setParameter("uuids", uuids)
+                .getResultList()
+        );
+
+        int updated = 0;
+        String updateJpql =
+            "UPDATE DeletedConsumer d " +
+            "   SET d.consumerName      = :consumerName, " +
+            "       d.ownerId           = :ownerId, " +
+            "       d.ownerDisplayName  = :ownerDisplayName, " +
+            "       d.ownerKey          = :ownerKey, " +
+            "       d.principalName     = :principalName, " +
+            "       d.updated           = :now " +
+            "WHERE d.consumerUuid      = :uuid ";
+        Query updateQ = this.getEntityManager()
+            .createQuery(updateJpql);
+
+        for (Object[] r : rows) {
+            String uuid = (String) r[1];
+            if (!existingUuids.contains(uuid)) {
+                continue;
+            }
+
+            updateQ.setParameter("consumerName", (String) r[5])
+                .setParameter("ownerId", (String) r[2])
+                .setParameter("ownerDisplayName", (String) r[3])
+                .setParameter("ownerKey", (String) r[4])
+                .setParameter("principalName", principalName)
+                .setParameter("now", Util.toDate(now))
+                .setParameter("uuid", uuid);
+            updated += updateQ.executeUpdate();
+            uuids.remove(uuid);
+        }
+
+        int inserted = 0;
+        if (!uuids.isEmpty()) {
+            String insertJpql =
+                "INSERT INTO DeletedConsumer (id, created, updated, consumerUuid, ownerId, " +
+                "ownerDisplayName, ownerKey, principalName, consumerName) " +
+                "SELECT c.id, :now, :now, c.uuid, c.ownerId, o.displayName, o.key, :principalName, c.name " +
```

Comment:
There's short-hand for now() as a constant/function that can be used so it doesn't need to be parameterized like this. The only reason we'd want to parameterize the timestamps is to ensure every entry uses the exact same timestamp, but in such a case you'd need to also update the parameter to use a fixed value rather than one which is generated repeatedly as is done on line 204.